src/index.ts

```diff
@@ -1,32 +1,50 @@
 // Import the express in typescript file
 import express from 'express';
 import { trade_defund } from './defund';
+import { Request, Response } from "express";
 
 // Initialize the express engine
 const app: express.Application = express();
 
 // Take a port 3000 for running server.
 const port: number = 3000;
-
+app.use(express.urlencoded({ extended: true }))
 // Handling '/' Request
-app.get('/', async (_req, _res) => {
-	const fundAddress = '';
-	const fromToken = '';
-	const toToken = '';
-	const slippage = 1; // 1%
-	const share = 10; // 10%
-	const network = '';
-	const options = {
-		gasLimit: 1000000,
-		// gasPrice: 60, // 60 gwei
-	} // optional
-	const tx = await trade_defund(fundAddress, fromToken, toToken, slippage, share, network, options);
-	_res.send(tx);
-});
-
+app.get("/trade", async (req: Request, res: Response) => {
+  try {
+    let protocol: string = 'defund';
+    if (req.query.protocol) protocol = req.query.protocol as string;
+    const assetA = req.query.from as string;
+    const assetB = req.query.to as string;
+    const share = req.query.share as string;
+    const amount = req.query.amount as string;
+    const slippage = req.query.slippage as string;
+    const poolAddress = req.query.pool as string;
+    const network = req.query.network as string
+    const options = {
+            gasLimit: 1000000,
+            gasPrice: 60, // 60 gwei
+    }
```

Comment:
maybe add optional comment is better

Reply:
what you mean ?

Reply:
I mean that add a comment for options param is optional

Reply:
Ok